TTHAnalysis/python/tools/higgsRecoTTH.py

```diff
@@ -29,52 +33,164 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+                  #print "the mother of this quark is W+ or W-"
+                  if abs(genpar[genpar[part.genPartIdxMother].genPartIdxMother].pdgId) == 25:
+                     #print "the mother of this W is a Higgs"
+                     QFromWFromH.append(part)
+            if abs(part.pdgId) in range (11,18) and part.status == 1:
+               #print "it is a lepton"
+               if part.genPartIdxMother >= 0 and abs(genpar[part.genPartIdxMother].pdgId) == 24: 
+                  #print "the mother of this lepton is W+ or W-"
+                  if abs(genpar[genpar[part.genPartIdxMother].genPartIdxMother].pdgId) == 25:
+                     #print "the mother of this W is a Higgs"
+                     LFromWFromH.append(part)
+            if abs(part.pdgId) in range (1,8):
+               #print "it is a quark"
+               if part.genPartIdxMother >= 0 and abs(genpar[part.genPartIdxMother].pdgId) == 24: 
+                  #print "the mother of this quark is W+ or W-"
+                  if abs(genpar[genpar[part.genPartIdxMother].genPartIdxMother].pdgId) == 6:
+                     #print "the mother of this W is a Top"
+                     QFromWFromT.append(part)
+            if abs(part.pdgId) in range (11,18) and part.status == 1:
+               #print "it is a lepton"
+               if part.genPartIdxMother >= 0 and abs(genpar[part.genPartIdxMother].pdgId) == 24: 
+                  #print "the mother of this lepton is W+ or W-"
+                  if abs(genpar[genpar[part.genPartIdxMother].genPartIdxMother].pdgId) == 6:
+                     #print "the mother of this W is a Top"
+                     LFromWFromT.append(part)
+    # loop over gen jets 
+    # ------------------ 
+        #for jet in genjet:
+            #if not jet.partonFlavour == 5 and not jet.partonFlavour == -5: #TODO that excludes b-jets but it is not necessary
+            #if jet.p4().Pt() > 30 and abs(jet.p4().Eta()) < 2.5:  # bit extreme cuts, I think supposed to be 24 and 2.4
+               #gengoodJets.append(jet)
+               #print "jet flavour = " + str(jet.partonFlavour) + " and mass = " + str(jet.p4().M()) + " GeV and pT = " + str(jet.p4().Pt())
+        #MET_pt   = getattr(event,"MET_pt")
+        #mhtJet25 = getattr(event,"mhtJet25_Recl")
+        nleps    = getattr(event,"nLepGood")
+        nFO      = getattr(event,"nLepFO"+self.label)
+        ileps    = getattr(event,"iLepFO"+self.label)
+        leps     = Collection(event,"LepGood","nLepGood")
+        lepsFO   = [leps[ileps[i]] for i in xrange(nFO)]
+        jets     = [x for x in Collection(event,"JetSel"+self.label,"nJetSel"+self.label)]
+        ret      = {} 
 
         for var in self.systsJEC:
             score = getattr(event,"BDThttTT_eventReco_mvaValue%s"%self.systsJEC[var])
-            
+            #print score
             candidates=[]
-            
             if score>self.cut_BDT_rTT_score:
-
+                #if not len(jets) >= 4: continue 
```

Comment:
That was a debug line, no? Please remove

Reply:
removed